x-conduit/src/X/Control/Monad/Trans/Resource.hs

```diff
@@ -0,0 +1,23 @@
+{-# OPTIONS_GHC -fno-warn-orphans #-}
+module X.Control.Monad.Trans.Resource where
+
+import Control.Monad.Trans.Class (lift)
+import Control.Monad.Trans.Resource (MonadResource, liftResourceT)
+import Control.Monad.Trans.Either (EitherT)
+
+-- | This instance is useful when adding errors to the monad used for some conduit processing
+-- when resources, like files, are involved. For example:
+--
+-- foo :: MonadResource m => ConduitM () Void (EitherT Text m) [ByteString]
+-- foo = CB.sourceFile "src/Main.hs"
+--  =$= CB.lines
+--  -- filter some lines
+--  =$= CL.map parse
+--  -- collect errors in EitherT Text m
+--  =$= CL.mapM hoistEither
+--  =$= CL.consume
+--
+-- parse :: ByteString -> Either Text ByteString
+-- parse _ = Left ":("
+instance MonadResource m => MonadResource (EitherT e m) where
```

Comment:
Don't have a compiler in front of me, but the simplest way I can see is that given the dependencies already there you should be able to just use ExceptT for the conduit code and if there is `EitherT` that needs to be composed with on the outside can just wrap it back up. My objection isn't to using MonadResource, it is just to adding an orphan, especially in a library that a _lot_ of code will have to use for other reasons is very risky from my perspective. I would prefer to avoid the orphans at all cost, but if we do want to have one, then they should be in thier own library so there is no risk of getting stuck in a "we need to depend on x and <something else>, but they have conflicting orphans" situation. If it is a sep library you can normally control that a lot better.